tasks/kernel_matrix_testing/README.md

```diff
@@ -1,425 +1,564 @@
+    *   [kmt.build](#kmtbuild)
+    *   [kmt.test](#kmttest)
+    *   [kmt.pause](#kmtpause)
+    *   [kmt.resume](#kmtresume)
+    *   [kmt.destroy-stack](#kmtdestroy-stack)
+    *   [kmt.config-ssh-key](#kmtconfig-ssh-key)
+    *   [kmt.explain-ci-failure](#kmtexplain-ci-failure)
+*   [Advanced Usage](#advanced-usage)
+    *   [Connecting to VMs](#connecting-to-vms)
+    *   [CI Integration](#ci-integration)
+    *   [Using with External ("Alien") VMs](#using-with-external-alien-vms)
+*   [Troubleshooting](#troubleshooting)
+    *   [macOS Network Issues](#macos-network-issues)
 
 ## Overview
 
-The Kernel Matrix Testing system is a new approach for testing system-probe. It uses libvirt and qemu to launch pre-provisioned VMs over a range of distributions and kernel versions. These VMs are used for running the test suite of system-probe.
+The Kernel Matrix Testing (KMT) system is a new approach for testing `system-probe`. It uses `libvirt` and `qemu` to launch pre-provisioned VMs over a range of distributions and kernel versions. These VMs are used for running the `system-probe` test suite.
 
 Developers can check out [this](https://datadoghq.atlassian.net/wiki/spaces/EBPFTEAM/pages/3278832713/Developer+Documentation) confluence page for more details about the system.
 
-This file will document the invoke tasks provided to easily manage the lifecycle of the VMs launched using this system.
+This document covers the `invoke` tasks provided to manage the lifecycle of the VMs.
 
-The system works on the concept of `stacks`. A `stack` is a collection of VMs, both local and remote. A `stack` is given a unique name by the user.
-If no stack name is provided, a name is generated automatically from the current branch name. This allows the developers to couple `stacks` with their git workflow.
-A `stack` may be:
-
-- Created
-- Configured
-- Launched
-- Paused
-- Resumed
-- Destroyed
-
-> All subsequent commands are assumed to be executed from the root directory of the datadog-agent repository.
+> All commands are assumed to be executed from the root directory of the `datadog-agent` repository.
 
 ## Dependencies
 
-All dependencies are installed by the `kmt.init` command, as defined [below](#initializing-the-environment).
+All dependencies are installed by the `kmt.init` command. See [kmt.init](#kmtinit) for details.
 
-> For macOS users: Internet Sharing might need to be enabled for the networking to work properly. Enable it in System Settings -> General -> Sharing if you find problems with the VM networks. It does not matter which interface you enable it on, as long as it is enabled and the connection being shared is the one you use for Internet connection. We'd also appreciate it if you reported it to the eBPF platform team, as it's not clear still whether Internet Sharing needs to be enabled at some point or not.
+For macOS users, see the [macOS Network Issues](#macos-network-issues) section in troubleshooting.
 
-## Getting started
+## Quick Start
```

Comment:
I think the quick start section is too long. This section should just contain 1 recommended approach for using KMT. This means
- Initializing with `inv kmt.init`
- using local VMs
- relying on KMT to generate the stack name. So no `--stack` parameter is passed.
- Using `inv kmt.test` to interact with the VMs.


All other features are power-user features, which should be explained below in the documentation in detail.

Reply:
That's a good point. I agree that the Quick Start section should be as streamlined as possible for new users.

My goal was to present a complete, end-to-end example of the most common development workflow.

Based on your feedback, I've simplified the examples in the guide to better reflect the "happy path." I've removed the explicit `--stack` parameter to encourage the default behavior of using the git branch name.

The Quick Start for Local VMs now presents the following essential steps for a typical session:

- kmt.init
- kmt.gen-config
- kmt.launch-stack
- kmt.status
- kmt.test (with SSH as an optional step for debugging)
- kmt.destroy-stack

I believe this structure will be a good compromise.